README.md

```diff
@@ -1,2 +1,61 @@
 # pyweek36
 PyWeek 36
+
+
+
+# Mechanics
+
+Side-scrolling 2D platform game over a grid system. Each grid cell contains one of three types of blocks:
+
+1. Air; the player can walk/fall through.
+2. Solid Block; the player collides with and thus can walk over, but not through.
+3. Dark Matter Block; usually behaves as Air.
```

Comment:
I think it would benefit more from a couple more block types to help with level design:

- Dark matter source: where the spread originates from, and cannot be converted 
- Permenant solid blocks: dark matter cannot spread into them, can be used for persistent platforms/walls
- Maybe a converted solid block? For blocks that were converted from dark matter that decays faster than normal solid blocks (as a compromise between solid blocks and solidified dark matter)

Reply:
I like the first two items, but don't really understand the third. 

I think number one is a good simple way to prevent the player from being able to eradicate Dark Matter from a level. Without it, we'd have to always have in mind during design to put in more spread than the player can fight off, which I think can be hard.

As per number two, I think it can be interesting but not for any game objectives. If, for example, the player is to  jump to the top while running away from spread so that he doesn't fall through it, then introducing a Permanent Solid Block would give the player a block to chill on eternally (and not fall). Then we'd have to introduce another loss condition.

Reply:
I also think we should think of these blocks as more of an attribute of the block, it's up to our collective art skills mostly but we can make some different sprites to keep the levels looking fresh.

Reply:
so the `air blocks` are blocks on which players can stand on but after a set amount of time it disappears?

Reply:
> I also think we should think of these blocks as more of an attribute of the block, it's up to our collective art skills mostly but we can make some different sprites to keep the levels looking fresh.

Right, that's why those are block types. We can have several blocks which are of the type Solid Block, like Stone or Grass.

Reply:
> I like the first two items, but don't really understand the third.

We may want solid blocks converted from dark matter to covnert back into dark matter faster than normal solid blocks (or just make them disappear after a fixed amount of time together). I think this would be best represented with a separate block type

> As per number two, I think it can be interesting but not for any game objectives. If, for example, the player is to jump to the top while running away from spread so that he doesn't fall through it, then introducing a Permanent Solid Block would give the player a block to chill on eternally (and not fall). Then we'd have to introduce another loss condition.

I was mainly thinking of using them for walls to prevent the player from being able to go through any wall by just waiting for the dark matter to spread to it (which would be hard/annoying to design against). It would also be useful to control where the dark matter can spread to without having gaps everywhere.

Being able to stay on one indefinitely is definitely an issue though. I think having small platforms made of these here and there as sort of checkpoints would be fine but we would definitely want a way to prevent abusing them (maybe another block that becomes deadly when infected rather than disappearing? although that would kinda break the theme)

Reply:
A couple more tile ideas:
- Tiles that you can shoot through but not walk through (self explanatory)
- Tiles in the background that dark matter can spread to/from, this will allow dark matter get to floating platforms too

Reply:
I don't think an undecayable block is really necessary, but we can decide to add it in later if we need.

Reply:
> A couple more tile ideas:
> 
> * Tiles that you can shoot through but not walk through (self explanatory)
> * Tiles in the background that dark matter can spread to/from, this will allow dark matter get to floating platforms too

oh, I actually love number two. I think it's important actually.

Reply:
> Tiles that you can shoot through but not walk through (self explanatory)

I don't see much gain from this, so I'd leave it for later consideraiton.

Reply:
I've updated the mechanics to incorporate the concept of foreground and background grid systems, so that we can have spread reaching floating platforms

Reply:
> We may want solid blocks converted from dark matter to covnert back into dark matter faster than normal solid blocks (or just make them disappear after a fixed amount of time together). I think this would be best represented with a separate block type

I feel this is already covered by the Solidified Dark Matter block type. I'll elaborate on it a bit to allow for more creativity moving forward

Reply:
> > We may want solid blocks converted from dark matter to covnert back into dark matter faster than normal solid blocks (or just make them disappear after a fixed amount of time together). I think this would be best represented with a separate block type
> 
> I feel this is already covered by the Solidified Dark Matter block type. I'll elaborate on it a bit to allow for more creativity moving forward

Oh yeah I missed that. 'Solidified Dark Matter' makes more sense too.